### ISSUE TITLE
[Feature] Add support for new mdblist.com API

### ISSUE THREAD

**Issue by linaspurinis** [jurialmunkey/plugin.video.themoviedb.helper#1296]
### Describe the feature that you'd like

Migration to new mdblist.com API: api.mdblist.com

### Why is this feature needed? What problem does it solve?

Currently, mdblist.com/api is limited in features and requires a single call per item to get media information. The new API supports batch media retrieval, and it is possible to get all media information with a single API call: https://mdblist.docs.apiary.io/#reference/0/media-info-batch/get-media-info-batch
I'm not too familiar with plugin work, so I don't think I can rewrite the code myself without proper testing. But I can help explain the API and help with code if needed. Thank you.

### Checklist

- [X] I have searched the issues and this feature has not been requested.
- [X] I have updated TMDbHelper to the most recent version.

**Comment by jurialmunkey** [comment]
Is there any issue with continuing to use the old API?

I don't have much use for batching because I only ever retrieve media details for ratings and these are only retrieved on-demand for the currently focused item once the user stops moving (not retrieved while scrolling or any sort of preemptive readahead caching)

I assume that is a better approach than retrieving several hundred ratings in batch whenever a list is opened *just in case* they are used but where it is more likely they will never be seen or needed.

**Comment by linaspurinis** [comment]
I agree, but the old API is no longer updated. For example, features like sorting or adding new items to static lists will only be supported in the new API.

**Comment by jurialmunkey** [comment]
> I agree, but the old API is no longer updated. For example, features like sorting or adding new items to static lists will only be supported in the new API.

I will eventually migrate to the new one, I just wanted to make sure there wasn't any problem with continuing to use the old one for time being until I need to add those other features / have the time to migrate it over.

Just want to make sure I'm respecting the service is all!

**Comment by jurialmunkey** [comment]
Actually managed to find some time this morning to update to the new API and refactor a few things.
